### PR TITLE
refactor(vsc): change environment tree to native provider to enhance UX

### PR DESCRIPTION
**Pull request by tecton**
1. Postpone the data update time when refreshing tree view.
2. Leverage the progress indicator of VSCode like "clock" and running progress bar.
3. Remove the registration of environment tree in non-teamsfx project.

E2E TEST: N/A

### CODE REVIEW

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        const subscriptionNode = new SubscriptionNode(this.identifier, subscriptionInfo);
+        children.push(subscriptionNode);
+      }
+    }
+    return children;
+  }
+
+  public async getTreeItem(): Promise<vscode.TreeItem> {
+    const envInfo = await this.getCurrentEnvInfo(this.identifier);
+    const isSpfxProject = await isSPFxProject(ext.workspaceUri.fsPath);
```
Comment (qinezh) [inline packages/vscode-extension/src/treeview/environmentTreeItem.ts:68]: seems the `await` is unnecessary.

Comment (qinezh) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts:68]: do we need to check if `ext.workspaceUri` is valid here?

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    } else {
+      this.resourceGroupNode = undefined;
+      this.collapsibleState = vscode.TreeItemCollapsibleState.None;
+    }
+
+    return this;
+  }
+}
+
+class ResourceGroupNode extends DynamicNode {
```
Comment (qinezh) [inline packages/vscode-extension/src/treeview/environmentTreeItem.ts:219]: where's the collaboration related node?

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts (hunk 1 above)
Comment (tecton) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts]: Thanks, fixed this `await` code.
The `ext.workspaceUri` is guaranteed in `EnvironmentTreeViewProvider:reloadEnvironments()` so it should always be valid here.

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts (hunk 2 above)
Comment (tecton) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts:220]: Do you mean `generateCollaboratorNode()` and `generateCollaboratorWarningNode()`? I didn't find any usage so I just removed these 2 functions.
Other collaboration functions seem to be provided by "view item context" and they are updated in `extension.ts`.

File: packages/vscode-extension/src/treeview/environmentTreeViewProvider.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  public static getInstance() {
+    if (!EnvironmentTreeViewProvider.instance) {
+      EnvironmentTreeViewProvider.instance = new EnvironmentTreeViewProvider();
+    }
+    return EnvironmentTreeViewProvider.instance;
+  }
+
+  public async reloadEnvironments(): Promise<Result<Void, FxError>> {
+    if (!ext.workspaceUri) {
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/treeview/environmentTreeViewProvider.ts:34]: Maybe check whether it's teamsfx project here so we don't need to continue for non-teamsfx projects?

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts (hunk 2 above)
Comment (qinezh) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts:220]: @SLdragon @KennethBWSong may take a look at the usage of collaboration node.

Comment (SLdragon) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts:220]: Thanks, we don't add collaboration node to the tree view any more, so we can safely remove this two function

**Review by SLdragon — APPROVED** [review]

File: packages/vscode-extension/src/treeview/environmentTreeViewProvider.ts (hunk 3 above)
Comment (tecton) [reply packages/vscode-extension/src/treeview/environmentTreeViewProvider.ts]: Thanks, fixed.

File: packages/vscode-extension/src/treeview/environmentTreeItem.ts (hunk 1 above)
Comment (tecton) [reply packages/vscode-extension/src/treeview/environmentTreeItem.ts]: This makes sense and it's fixed.

**Review by qinezh — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**